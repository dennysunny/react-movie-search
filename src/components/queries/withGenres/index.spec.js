import { graphql } from 'react-apollo';
import React from 'react';
import { shallow } from 'enzyme';
import withGenres from '.';

describe('Queries/withGenres', () => {
  graphql.mockImplementation((query, config) => {
    const ownProps = { memberId: 1234, ssn: 123456 };
    const props = config.props({
      data: {
        genres: {
          id: 1,
          name: 'genrename',
        },
      },
      ownProps,
    });
    const options = config.options(ownProps);
    return jest.fn(Mock => <Mock options={options} {...props} />);
  });

  const MockComponent = props => ({ ...props });
  const Component = withGenres()(MockComponent);

  it('should render as expected', () => {
    const render = shallow(Component);
    expect(render).toMatchSnapshot();
  });
});
