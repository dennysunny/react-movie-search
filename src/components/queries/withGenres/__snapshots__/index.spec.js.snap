// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queries/withGenres should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MockComponent
    genres={
      Object {
        "id": 1,
        "name": "genrename",
      }
    }
    options={
      Object {
        "fetchPolicy": "cache-and-network",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "genres": Object {
      "id": 1,
      "name": "genrename",
    },
    "genresLoading": undefined,
    "options": Object {
      "fetchPolicy": "cache-and-network",
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "genres": Object {
        "id": 1,
        "name": "genrename",
      },
      "genresLoading": undefined,
      "options": Object {
        "fetchPolicy": "cache-and-network",
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
